namer/namer.cc

```diff
@@ -1508,7 +1499,18 @@ class SymbolDefiner {
                 deleteMethodViaFullNameHash(ctx, oldMethodHash);
             }
         }
+    }
 
+public:
+    SymbolDefiner(State state, const core::FoundDefinitions &foundDefs, optional<core::FoundDefHashes> oldFoundHashes)
+        : state(move(state)), foundDefs(foundDefs), oldFoundHashes(move(oldFoundHashes)) {}
+
+    SymbolDefiner::State beginIncrementalDefinition(core::MutableContext ctx) {
+        beginIncrementalDefinitionInternal(ctx);
+        return move(state);
+    }
+
+    void finishIncrementalDefinition(core::MutableContext ctx) {
         for (auto &method : foundDefs.methods()) {
```

Comment:
```suggestion
        // We have to defer defining "deletable" symbols until the "finish" phase of incremental
        // namer so that we don't delete and immediately re-enter a symbol (possibly keeping it
        // alive, if it had multiple locs at time of deletion) before SymbolDefiner has had a chance
        // to process _all_ files.
        for (auto &method : foundDefs.methods()) {
```